src/frontend/src/hooks/AuthHooks.ts

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react';
+import { Nullable } from 'types/types';
+import {ApiError } from './types';
+import { supabase } from 'main';
+import { Session } from '@supabase/supabase-js';
+
+export default function useAuth() {
+    const [isLoading, setLoading] = useState<boolean>(true);
+    const [session, setSession] = useState<Session | null>();
+    const [errors, _] = useState<Nullable<ApiError>>(null);
+
+    useEffect(() => {
+        setLoading(true);
+        const fetchSession = async () => {
+            supabase.auth.getSession().then(({ data: { session } }) => {
+                setSession(session);
+            });
+            setLoading(false);
+        };
+        fetchSession();
+    }, []);
+
+    const logout = async (session: Session) => {
+        supabase.auth.signOut();
```

Comment:
I'm confused, I would have assumed that `signOut` would require the current session to run (ie. session as a param). If not,

* can we remove the session param from our wrapper logout method?
* how does SupaBase signout the current session

> Although I say I'm confused here, I suspect what SupaBase does is introspect the session cookie and sign that user out. This makes sense from a single-tenant per-browser design standpoint